cmd/root.go

```diff
@@ -0,0 +1,106 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/github/hub/github"
+	"github.com/github/hub/ui"
+	"github.com/github/hub/utils"
+
+	"github.com/rgreinho/keeparelease/keeparelease"
+	"github.com/spf13/cobra"
+)
+
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "keeparelease",
+	Short: "Create beautiful GitHub releases.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Get the flags.
+		extract, err := cmd.Flags().GetBool("extract")
+		utils.Check(err)
+
+		// Read the last release information from the Changelog.
+		title, content, err := keeparelease.ReadChangelog()
+		if err != nil {
+			return err
+		}
+
+		// If extract only, simply display the content of the last release.
+		if extract {
+			fmt.Print(content)
+			return nil
+		}
+
+		// Inspect the local repo.
+		localRepo, err := github.LocalRepo()
+		utils.Check(err)
+		project, nil := localRepo.MainProject()
+		utils.Check(err)
+
+		// Prepare the Hub client.
+		gh := github.NewClient(project.Host)
+
+		// Prepare the release.
+		tag, err := cmd.Flags().GetString("tag")
+		params := &github.Release{
+			TagName: tag,
+			Name:    title,
+			Body:    content,
+		}
+
+		// Create the release.
+		var release *github.Release
+		release, err = gh.CreateRelease(project, params)
+		utils.Check(err)
+
+		// Upload assets.
+		assets, err := cmd.Flags().GetStringArray("attach")
+		uploadAssets(gh, release, assets)
+
+		return nil
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	// rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.keeparelease.yaml)")
+	rootCmd.Flags().BoolP("extract", "x", false, "Only extract the last release information")
+	rootCmd.Flags().StringP("tag", "t", "", "Use a specific tag")
+	rootCmd.Flags().StringArrayP("attach", "a", []string{}, "Specify the assets to include into the release")
```

Comment:
This is less about the code than about the tool itself, but I have no idea what these assets might be.